### ISSUE TITLE
Tpetra:  Why remove instead of deprecate getLocalRowView?

### ISSUE THREAD

**Issue by tcfisher** [trilinos/Trilinos#9082]
## Question
@trilinos/tpetra
@kddevin 
@ccober6 

The commit, https://github.com/trilinos/Trilinos/commit/34a07602a38f585b1b040f056c58af9f0624e7d4, has broken SPARC because SPARC uses getLocalRowView from TpetraBlockVector and TpetraBlockCrs. A lot of code was moved to deprecated when new interfaces were added. Why wasn't this made deprecated instead of removed? 

What is the correct translation to transition from getLocalRowView to getBlock?

Thanks,

Travis

**Comment by jhux2** [comment]
@trilinos/tpetra

**Comment by kddevin** [comment]
Removing without deprecation was likely a mistake.  I'll restore it this afternoon unless there is a serious problem that led to the removal.  Thanks for the report; sorry for the inconvenience.
@tcfisher

**Comment by kddevin** [comment]
@tcfisher #9084 restores the missing functions.  Note, however, that these functions will be deprecated.
Instead of saying
```
scalar *x;
mv.getLocalRowView(row, col, x);
```
you should do
```
auto x = mv.getLocalBlockHost(row, col, Access::ReadOnly);
```
or
```
auto x = mv.getLocalBlockHost(row, col, Access::ReadWrite);
```
The access tag says how you intend to use the returned view -- as read-only (const) data or read-write (non-const) data.

Deprecated function getLocalBlock, getLocalRowView and getGlobalRowViews were returning host data.  Please let me know if you need device versions of getLocalBlock.  Or if you have suggestions for better block interfaces, let us know.

Note that similar changes will happen for BlockCrsMatrix.  @kyungjoo-kim is working on BlockCrsMatrix now.

**Comment by tcfisher** [comment]
Thanks @kddevin, we will try to transition to the new interface soon. We were only using this on the host.

**Comment by kddevin** [comment]
Fixed by #9084
